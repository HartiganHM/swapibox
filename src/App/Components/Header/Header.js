import React from 'react';
import PropTypes from 'prop-types';
import './Header.css';

const Header = ({ crawlData }) => {
  const romanNumeral = {
    1: 'I',
    2: 'II',
    3: 'III',
    4: 'IV',
    5: 'V',
    6: 'VI',
    7: 'VII'
  };

  return (
    <div className="Header">
      <div className="crawl">
        <div className="title">
          <span>Episode {romanNumeral[crawlData.episodeNum]}</span>
          <span>{crawlData.episodeTitle}</span>
        </div>
        <p>{crawlData.openingCrawl}</p>
      </div>
    </div>
  );
};

Header.propTypes = {
  crawlData: PropTypes.object
};

export default Header;
