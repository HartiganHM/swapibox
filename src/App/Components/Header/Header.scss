@import '../../../variables.scss';

.Header {
    background: url('../../../Images/stars.jpg');
    background-size: contain;
    color: $accent-color;
    display: flex;
    flex: 1;
    font: {
        family: $accent-font;
        size: 500%;
        weight: 600;
    }
    justify-content: center;
    letter-spacing: 0.6rem;
    line-height: 150%;
    overflow: hidden;
    padding:$padding-standard;
    perspective: 40rem;
    text-align: justify;

    .crawl {
        position: relative;
        top: -10rem;
        transform-origin: 50% 100%;
        animation: crawl 90s linear;
        animation-iteration-count: infinite;

        .title {
            display: flex;
            flex-direction: column;
            text-align: center;
        }
    }
}

@keyframes crawl {
    0% {
        top: 0;
        transform: rotateX(20deg) translateZ(0);
    }

    100% {
        top: -600rem;
        transform: rotateX(25deg) translateZ(-250rem);
    }
}