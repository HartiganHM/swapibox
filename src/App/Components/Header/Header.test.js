import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header Tests', () => {
  let renderedHeader;
  let mockCrawlData;

  beforeEach(() => {
    mockCrawlData = {
      episodeNum: 7,
      episodeTitle: 'Star Trek N Stuff',
      openingCrawl: 'Me crawl!'
    };
    renderedHeader = shallow(<Header crawlData={mockCrawlData} />);
  });

  it('Should exist', () => {
    expect(renderedHeader).toBeDefined();
  });

  it('Should match the snapshot', () => {
    expect(renderedHeader).toMatchSnapshot();
  });

  it('Should convert the episode number to a Roman numeral', () => {
    const mockEpisodeNumber = renderedHeader.find('span').first();
    const expectedText = 'Episode VII';

    expect(mockEpisodeNumber.text()).toEqual(expectedText);
  });
});
