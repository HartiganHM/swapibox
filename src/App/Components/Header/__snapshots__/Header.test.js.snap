// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    crawlData={
        Object {
            "episodeNum": 7,
            "episodeTitle": "Star Trek N Stuff",
            "openingCrawl": "Me crawl!",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <div
          className="crawl"
>
          <div
                    className="title"
          >
                    <span>
                              Episode 
                              VII
                    </span>
                    <span>
                              Star Trek N Stuff
                    </span>
          </div>
          <p>
                    Me crawl!
          </p>
</div>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
>
              <span>
                            Episode 
                            VII
              </span>
              <span>
                            Star Trek N Stuff
              </span>
</div>,
            <p>
              Me crawl!
</p>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Episode 
                  VII
</span>,
                <span>
                  Star Trek N Stuff
</span>,
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Episode ",
                    "VII",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Episode ",
                  "VII",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Star Trek N Stuff",
                },
                "ref": null,
                "rendered": "Star Trek N Stuff",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Me crawl!",
            },
            "ref": null,
            "rendered": "Me crawl!",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <span>
                                    Episode 
                                    VII
                        </span>
                        <span>
                                    Star Trek N Stuff
                        </span>
            </div>
            <p>
                        Me crawl!
            </p>
</div>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <span>
                                Episode 
                                VII
                </span>
                <span>
                                Star Trek N Stuff
                </span>
</div>,
              <p>
                Me crawl!
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Episode 
                    VII
</span>,
                  <span>
                    Star Trek N Stuff
</span>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      "VII",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    "VII",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Star Trek N Stuff",
                  },
                  "ref": null,
                  "rendered": "Star Trek N Stuff",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Me crawl!",
              },
              "ref": null,
              "rendered": "Me crawl!",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
